packages/datadog-instrumentations/src/mariadb.js

```diff
@@ -8,82 +8,56 @@ const startCh = channel('apm:mariadb:query:start')
 const finishCh = channel('apm:mariadb:query:finish')
 const errorCh = channel('apm:mariadb:query:error')
 
-function wrapConnectionAddCommand (addCommand) {
-  return function (cmd) {
-    if (!startCh.hasSubscribers) return addCommand.apply(this, arguments)
+function wrapCommandStart (start) {
+  return function () {
+    if (!startCh.hasSubscribers) return start.apply(this, arguments)
 
-    const asyncResource = new AsyncResource('bound-anonymous-fn')
-    const name = cmd && cmd.constructor && cmd.constructor.name
-    const isCommand = typeof cmd.start === 'function'
-    const isQuery = isCommand && (name === 'Execute' || name === 'Query')
-
-    // TODO: consider supporting all commands and not just queries
-    cmd.start = isQuery
-      ? wrapStart(cmd, cmd.start, asyncResource, this.opts)
-      : bindStart(cmd, cmd.start, asyncResource)
-
-    return asyncResource.bind(addCommand, this).apply(this, arguments)
-  }
-}
+    const callbackResource = new AsyncResource('bound-anonymous-fn')
 
-function bindStart (cmd, start, asyncResource) {
-  return asyncResource.bind(function (packet, connection) {
-    if (this.resolve) {
-      this.resolve = asyncResource.bind(this.resolve)
-    }
+    const resolve = callbackResource.bind(this.resolve)
+    const reject = callbackResource.bind(this.reject)
 
-    if (this.reject) {
-      this.reject = asyncResource.bind(this.reject)
-    }
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
 
-    return start.apply(this, arguments)
-  }, cmd)
+    shimmer.wrap(this, 'resolve', function wrapResolve () {
```

Comment:
Is there a way we can patch the prototype instead? Best to avoid patching on every call, if we can. 🤔

Reply:
No because these are assigned in the [constructor](https://github.com/mariadb-corporation/mariadb-connector-nodejs/blob/f6c0aceee29236a195a9694a07f6ebf026f12656/lib/cmd/command.js#L14-L23).